### DISCUSSION TITLE
Programmaticaly monitoring

### DISCUSSION THREAD

**Discussion by pierremestrez1976** [jackaudio/jack2#769] [Q&A]
Hello,
I would like to create a program that wait for a Jack Router client to connect then connect its port to other clients.
I'm thinking about programmaticaly (c++) monitor for a Jack Router client to connect then establish connection.
How to properly wait for a client to be connected ?
Thanks,
Pierre

**Comment by falkTX** [comment]
Have a look at https://github.com/jackaudio/jack2/blob/develop/tools/evmon.c for some examples.
The full API is documented at https://jackaudio.org/api/

**Reply by pierremestrez1976** [reply]
Thank you ! Exactly the couple of line I was missing.
